@here/harp-mapview/lib/Utils.ts

```diff
@@ -239,9 +239,143 @@ export namespace MapViewUtils {
+                const rotQuaternion = cache.quaternions[0].setFromAxisAngle(
+                    rotPlaneNorm,
+                    tangentAngle
+                );
+                const newLookAt = cameraToGround.clone().applyQuaternion(rotQuaternion);
+                rayCaster.set(cameraPos, newLookAt);
+                result = rayCaster.ray.intersectPlane(groundPlane, new THREE.Vector3());
+            } else {
+                assert(false, "Orthogonal camera is not yet supported");
+            }
+        } else if (projection.type === ProjectionType.Spherical) {
+            // Tolerance may be adjusted to not exactly hit the sphere tangent.
+            const sphericalTolerance = THREE.MathUtils.degToRad(0.0);
+            const cameraToOrigin = cache.vector3[2].copy(camera.position).negate();
+            const r = EarthConstants.EQUATORIAL_RADIUS;
+            const d = cameraToOrigin.length();
+            // Normalize cameraToOrigin vector
+            cameraToOrigin.normalize();
+            if (camera instanceof THREE.PerspectiveCamera) {
+                // Calculate angle between surface normal (below camera position) and tangent.
+                const tangentAngle = Math.asin(r / d);
+
+                /* Alternative approach
+                // Calculate angle between look at and surface normal(below camera position)
+                const cameraPitch = getCameraPitch(cameraToOrigin, cameraLookAt);
+                console.log("Pitch angle: ", THREE.MathUtils.radToDeg(cameraPitch));
+                // Calculate angle between camera eye vector and tangent.
+                const diffAngle = tangentAngle - cameraPitch;
+                console.log("Diff angle: ", THREE.MathUtils.radToDeg(diffAngle));
+                // Use tangent based look at distance if horizon is within field of view
+                const tangentDist = getTangentBasedLookAtDist(d, r, diffAngle);
+                // Above be used to simply multiply look at by tangent distance without raycasting
+                */
+                // Calculate the plane of rotation as cross product of look at and camera to origin
+                const rotPlaneNorm = cameraToOrigin
+                    .clone()
+                    .cross(cameraLookAt)
+                    .normalize();
+                // Find rotation quaternion around the rotation plane normal.
+                const rotQuaternion = cache.quaternions[0].setFromAxisAngle(
+                    rotPlaneNorm,
+                    tangentAngle - sphericalTolerance
+                );
+                // Finally find the gazing (tangent based) look at vector.
+                const newLookAt = cameraToOrigin.applyQuaternion(rotQuaternion);
+                rayCaster.set(cameraPos, newLookAt);
+                result = rayCaster.ray.intersectSphere(groundSphere, new THREE.Vector3());
+            } else {
+                assert(false, "Orthogonal camera is not yet supported");
```

Comment:
And most likely will never be. Please remove the if and change the input type.

Reply:
Hehhe... OK!